instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractorTest.java

```diff
@@ -200,6 +205,78 @@ void normal() {
             entry(NETWORK_PEER_PORT, 456L));
   }
 
+  @ParameterizedTest
+  @ArgumentsSource(StripUrlArgumentSource.class)
+  void stripBasicAuthTest(String url, String expectedResult) {
+    Map<String, String> request = new HashMap<>();
+    request.put("urlFull", url);
+
+    System.setProperty(
+        "otel.instrumentation.http.client.experimental.redact-sensitive-parameters", "true");
+    AttributesExtractor<Map<String, String>, Map<String, String>> extractor =
+        HttpClientAttributesExtractor.create(new TestHttpClientAttributesGetter());
+
+    AttributesBuilder attributes = Attributes.builder();
+    extractor.onStart(attributes, Context.root(), request);
+
+    assertThat(attributes.build()).containsOnly(entry(URL_FULL, expectedResult));
+  }
+
+  static final class StripUrlArgumentSource implements ArgumentsProvider {
+
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+      return Stream.of(
+          arguments("https://user1:secret@github.com", "https://REDACTED:REDACTED@github.com"),
+          arguments(
+              "https://user1:secret@github.com/path/",
+              "https://REDACTED:REDACTED@github.com/path/"),
+          arguments(
+              "https://user1:secret@github.com#test.html",
+              "https://REDACTED:REDACTED@github.com#test.html"),
+          arguments(
+              "https://user1:secret@github.com?foo=b@r",
+              "https://REDACTED:REDACTED@github.com?foo=b@r"),
+          arguments(
+              "https://user1:secret@github.com/p@th?foo=b@r",
+              "https://REDACTED:REDACTED@github.com/p@th?foo=b@r"),
+          arguments("https://github.com/p@th?foo=b@r", "https://github.com/p@th?foo=b@r"),
+          arguments("https://github.com#t@st.html", "https://github.com#t@st.html"),
+          arguments("user1:secret@github.com", "user1:secret@github.com"),
+          arguments("https://github.com@", "https://github.com@"),
+          arguments(
```

Comment:
This test was (probably inadvertly) removed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9925.

The test is added again with new test cases.